simd/src/Kokkos_SIMD_AVX2.hpp

```diff
@@ -784,6 +813,39 @@ class simd<std::int64_t, simd_abi::avx2_fixed_size<4>> {
       _mm256_add_epi64(static_cast<__m256i>(lhs), static_cast<__m256i>(rhs)));
 }
 
+// Shift right arithmetic for 64bit packed ints is not availalbe in AVX2
+// KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION
+// simd<std::int64_t, simd_abi::avx2_fixed_size<4>> operator>>(
+//     simd<std::int64_t, simd_abi::avx2_fixed_size<4>> const& lhs,
+//     int rhs) noexcept {
+//   return simd<std::int64_t, simd_abi::avx2_fixed_size<4>>(
+//       _mm256_srai_epi64(static_cast<__m256i>(lhs), rhs));
+// }
+// KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION
+// simd<std::int64_t, simd_abi::avx2_fixed_size<4>> operator>>(
+//     simd<std::int64_t, simd_abi::avx2_fixed_size<4>> const& lhs,
+//     simd<int, simd_abi::avx2_fixed_size<4>> const& rhs) noexcept {
+//   return simd<std::int64_t, simd_abi::avx2_fixed_size<4>>(
+//       _mm256_srav_epi64(static_cast<__m256i>(lhs),
+//       _mm256_cvtepi32_epi64(static_cast<__m128i>(static_cast<__m128i>(rhs))));
+// }
+
+KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION
+simd<std::int64_t, simd_abi::avx2_fixed_size<4>> operator<<(
+    simd<std::int64_t, simd_abi::avx2_fixed_size<4>> const& lhs,
+    int rhs) noexcept {
+  return simd<std::int64_t, simd_abi::avx2_fixed_size<4>>(
+      _mm256_slli_epi64(static_cast<__m256i>(lhs), rhs));
+}
+KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION
+simd<std::int64_t, simd_abi::avx2_fixed_size<4>> operator<<(
+    simd<std::int64_t, simd_abi::avx2_fixed_size<4>> const& lhs,
+    simd<int, simd_abi::avx2_fixed_size<4>> const& rhs) noexcept {
+  return simd<std::int64_t, simd_abi::avx2_fixed_size<4>>(
+      _mm256_sllv_epi64(static_cast<__m256i>(lhs),
+                        _mm256_cvtepi32_epi64(static_cast<__m128i>(rhs))));
```

Comment:
Please explain why we need this.

Reply:
Sorry I meant the `_mm256_cvtepi32_epi64(static_cast<__m128i>(rhs))`

Reply:
`_mm256_sllv_epi64` takes in ` __m256i` as its arguments. Because `rhs` is `int32_t` simd typed (`__m128i`), it needs to be extended to 64-bit int simd type. So `_mm256_cvtepi32_epi64` is used to sign extend `rhs` to `__m256i`.

I think the function signature that expects `rhs` as `simd<int32_t, abi>` could be up for a discussion for a change. [Kokkos document](https://kokkos.github.io/kokkos-core-wiki/API/simd/simd.html) has `simd operator>>(const simd& lhs, const simd& rhs)` for the operator, and I think the type of `rhs` may be ambiguous as is.

Reply:
I am pretty sure they just want the same operator on both sides? https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2023/p1928r4.pdf does not have mixed operator versions. i.e. return type, lhs and rhs are all the same. Now you probably still can pass another thing in and rely on conversion. but it does not explicitly support it. Should we? @ldh4 @ibaned @dalg24 So essentially this should be a signature 
```
simd<std::int64_t, simd_abi::avx2_fixed_size<4>> operator<<(
    simd<std::int64_t, simd_abi::avx2_fixed_size<4>> const& lhs,
    simd<std::int64_t, simd_abi::avx2_fixed_size<4>> const& rhs)
```

Reply:
Modified shift operators so that its return type, `lhs` and `rhs` be the same simd type in [63c8055](https://github.com/kokkos/kokkos/pull/6109/commits/63c80558527bb520f6b2fb6c5ae605c94b8c362d).
Shift operators that take in an int as `rhs` still remain.